File: api/src/main/java/net/onelitefeather/pandorascluster/api/notification/StaffNotification.java
package net.onelitefeather.pandorascluster.api.notification;

import net.onelitefeather.pandorascluster.api.enums.EntityCategory;
import net.onelitefeather.pandorascluster.api.land.Land;

import java.util.EnumMap;
import java.util.Map;

public abstract class StaffNotification {

    private static final Map<EntityCategory, Long> staffNotificationCooldown = new EnumMap<>(EntityCategory.class);

    public boolean canBeNotified(EntityCategory entityCategory) {
        return System.currentTimeMillis() >= staffNotificationCooldown.getOrDefault(entityCategory, 0L);
    }

    /**
     * @param entityCategory the entity category
     * @param cooldown the cooldown time in minutes
     */
    public void updateCooldown(EntityCategory entityCategory, Integer cooldown) {
        var time = cooldown != null ? cooldown : 1;
        staffNotificationCooldown.put(entityCategory, System.currentTimeMillis() + 1000L * 60 * time);
    }

    /**
     * @param land the land
     * @param entityCategory the entity category
     */
    public abstract void notifyEntitySpawnLimit(Land land, EntityCategory entityCategory);
}

File: api/src/main/kotlin/net/onelitefeather/pandorascluster/api/notification/StaffNotification.kt
package net.onelitefeather.pandorascluster.api.notification

import net.onelitefeather.pandorascluster.api.land.Land
import net.onelitefeather.pandorascluster.api.PandorasClusterApi
import net.onelitefeather.pandorascluster.api.enums.EntityCategory
import java.util.*

abstract class StaffNotification(private val pandorasClusterApi: PandorasClusterApi) {

    private val staffNotificationCooldown: MutableMap<EntityCategory, Long> = EnumMap(EntityCategory::class.java)

    /**
     * @param entityCategory the entity category
     * @return true if the cooldown is over
     */
    fun canBeNotified(entityCategory: EntityCategory) =
        System.currentTimeMillis() >= staffNotificationCooldown.getOrDefault(entityCategory, 0L)

    /**
     * @param entityCategory the entity category
     * @param cooldown the cooldown time in minutes
     */
    fun updateCooldown(entityCategory: EntityCategory, cooldown: Int?) {
        val time = cooldown ?: 1
        staffNotificationCooldown[entityCategory] = System.currentTimeMillis() + 1000 * 60 * time
    }

    /**
     * @param land the land
     * @param entityCategory the entity category
     */
    abstract fun notifyEntitySpawnLimit(land: Land, entityCategory: EntityCategory)

}